src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsJsonEncoding.cs

```diff
@@ -356,12 +356,27 @@ LaunchProfile ReadProfile(string name)
             ReadObject(property =>
             {
                 builder ??= ImmutableArray.CreateBuilder<(string Name, string Value)>();
-                builder.Add((property, ReadString()));
+                builder.Add((property, ReadEnvironmentVariableString()));
             });
 
             return builder?.ToImmutable() ?? ImmutableArray<(string Name, string Value)>.Empty;
         }
 
+        string ReadEnvironmentVariableString()
+        {
+            if (!reader.Read())
+            {
+                throw new JsonReaderException($"Cannot read string at {reader.LineNumber}.");
+            }
+
+            return reader.TokenType switch
+            {
+                JsonToken.String => (string)reader.Value!,
+                JsonToken.Date   => reader.Value!.ToString(),
+                _ => throw new JsonReaderException($"Expected string at {reader.LineNumber}:{reader.LinePosition} but got {reader.TokenType}.")
```

Comment:
I think you're just intending to read the raw string. It doesn't look like you actually care about the type of the original value. According to [this blog post](https://deliverystack.net/2021/05/15/migrating-c-from-newtonsoft-json-to-system-text-json-for-net-5/), the equivalent of [GetRawText()](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getrawtext?view=net-6.0) from System.Text.Json is simply doing [ToString](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Linq_JToken_ToString.htm) on the value.

Meaning, you don't need to parse the `TokenType`. Just do `reader.Value.ToString()` for anything.

Reply:
It seems that for environment variables converting everything to string makes sense.

Changes as suggested.